### ISSUE TITLE
manual configuration

### ISSUE THREAD

**Issue by drernie** [quiltdata/nf-quilt#185]
Right now we:

* use the URI to pass in all the information
* scrape the Params file to match the original URI with the generated QuiltPath

This is already brittle, and breaks completely for dynamically-generated URIs.

We should have the option to manually specify all the relevant parameters, and use those when available.
This is how it was done for the original [Seqera-built nf-quilt](https://github.com/nextflow-io/nf-quilt) plugin:
```
quilt {
  packageName = 'genomes/yeast'
  registry = 's3://seqera-quilt'
  message = 'My commit message'
  meta = [pipeline: 'nf-core/rnaseq']
  force = false
}
```

**Comment by w-osman-orion** [comment]
I am assuming this would be for the package the data is published to. If so, it would be great to have the additional following options:

- Skip metadata (i.e. use the existing metadata if the package is already created)
- Skip Readme creation
- include a parameter for the path within the quilt package

Also, I am not sure if this addresses the issue, as I was parsing the input parameters to dynamically create the URI's for both the inputs and outputs to the nf workflow.

**Comment by drernie** [comment]
You are right, it makes much more sense to have those application-specific configuration parameters inside your NextFlow configuration, rather than inside the URI.

The way this solves your dynamic URI problem is that right now nf-quilt tries to use the package name (from the path) to find the original output URI from the parameters file. Which fails because your inputs and outputs are from the same package are not in the parameters file.

This way:

1. nf-quilt can look at the config struct, rather than the params file, and
2. you can explicitly tell nf-quilt which output URI to use, and
3. you can simply embed the metadata in the config file (which is the actual bug you are facing ,right?)

**Comment by w-osman-orion** [comment]
Yes, that is the bug I am facing.

**Comment by drernie** [comment]
Hopefully this is fixed by 0.8.11.  Please let us know.

**Comment by drernie** [comment]
@w-osman-orion Note: these (by default) live in the nextflow.config, not the workflow, so they are somewhat "static" to the project.
